test/e2e/performanceprofile/functests/utils/profilesupdate/profile_update.go

```diff
@@ -77,3 +85,82 @@ func ApplyProfile(profile *performancev2.PerformanceProfile) error {
 	}
 	return nil
 }
+
+// WaitForTuningUpdating is waiting for the cluster to transition into tuning configuration update state
+// done by PerformanceProfile application.
+func WaitForTuningUpdating(ctx context.Context, profile *performancev2.PerformanceProfile) {
+	GinkgoHelper()
+	// In case we are on OCP, we can query the MCP to determine if the update has started.
+	if !hypershift.IsHypershiftCluster() {
+		performanceMCP, err := mcps.GetByProfile(profile)
+		Expect(err).ToNot(HaveOccurred())
+		testlog.Info("waiting for MCP starting to update")
+		mcps.WaitForCondition(performanceMCP, mcv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+		return
+	}
+
+	// On hypershift we can check the nodepool UpdatingConfig condition to determine if the update has started.
+	hostedClusterName, err := hypershift.GetHostedClusterName()
+	Expect(err).ToNot(HaveOccurred())
+	np, err := nodepools.GetByClusterName(ctx, testclient.ControlPlaneClient, hostedClusterName)
+	Expect(err).ToNot(HaveOccurred())
+	testlog.Infof("wait for node pool %q transition into update config state", client.ObjectKeyFromObject(np).String())
+	err = nodepools.WaitForUpdatingConfig(ctx, testclient.ControlPlaneClient, np.Name, np.Namespace)
+	Expect(err).ToNot(HaveOccurred())
+}
+
+// WaitForTuningUpdated is waiting for the cluster to come back from tuning configuration
+// done by PerformanceProfile application.
+// This is a lower lever function and should not be used directly.
+// Use PostUpdateSync instead
+func WaitForTuningUpdated(ctx context.Context, profile *performancev2.PerformanceProfile) {
+	GinkgoHelper()
+	// In case we are on OCP, we can query the MCP to determine if the update has completed.
+	if !hypershift.IsHypershiftCluster() {
+		performanceMCP, err := mcps.GetByProfile(profile)
+		Expect(err).ToNot(HaveOccurred())
+		testlog.Infof("waiting for MCP being updated")
+		mcps.WaitForCondition(performanceMCP, mcv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
```

Comment:
previously we were waiting on both performance mcp AND worker mcp:
now I can see its only performance mcp. did we drop that on purpose ?  
Tracking that from the new `PostUpdateSync`

Reply:
> previously we were waiting on both performance mcp AND worker mcp:
Could you please add a reference?
This function is new

Reply:
see https://github.com/openshift/cluster-node-tuning-operator/pull/1084/files#diff-decd79950ca0682eefd520e1171ff0d6c0af15779a02cc59fc71c4dbe1012bfaL68

Reply:
I missed this change:
1. This was added by @ffromani back in the time, maybe he remember the reason for waiting on both MCPs https://github.com/openshift/cluster-node-tuning-operator/commit/c5cf0bd33c5afae7695fd86d08c62aef269105d8#diff-decd79950ca0682eefd520e1171ff0d6c0af15779a02cc59fc71c4dbe1012bfa 
Usually we're not waiting for the worker MCP.
2. The tests are passing consistently. 

Anyway this is not an excuse, just stating the facts. I'll try to work something out on Sunday

Reply:
I don't remember the exact reason. I won't rule out it was a testing artifact (= workaround) which wasn't caught in review. I'd not be too worried about waiting for both if tests pass consistently.

Reply:
Then if it's agreed we can merge and evaluate later if needed

Reply:
yes, let's do this way